### PR TITLE
Add manual for delta node

### PR DESCRIPTION
**Pull request by albertpod**
Address #46

### CODE REVIEW

File: docs/src/manuals/delta-node.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+Based on the above, RxInfer.jl supports the following deterministic transformation scenarios:
+
+1. **Gaussian Nodes**: For delta nodes linked to strictly multivariate or univariate Gaussian distributions, the recommended methods are Linearization or Unscented transforms.
+2. **Exponential Family Nodes**: For the delta node connected to nodes from the exponential family, the CVI (Conjugate Variational Inference) is the method of choice.
+3. **Stacking Delta Nodes**: For scenarios where delta nodes are stacked, either Linearization or Unscented transforms are suitable.
+
+## Gaussian Case
+
+In the context of Gaussian distributions, we recommend either the `Linearization` or `Unscented` method for delta node approximation. The `Linearization` method provides a first-order approximation, while the `Unscented` method delivers a more precise second-order approximation. It's worth noting that while the `Unscented` method is more accurate, it also requires a little more computational resources, still pretty fast though. By default, delta node approximation in RxInfer.jl employs the `Unscented` method.
```
[the remark this answers is not in the document]
Comment (bvdmitri) [reply docs/src/manuals/delta-node.md:23]:
```suggestion
In the context of Gaussian distributions, we recommend either the `Linearization` or `Unscented` method for delta node approximation. The `Linearization` method provides a first-order approximation, while the `Unscented` method delivers a more precise second-order approximation. It's worth noting that while the `Unscented` method is more accurate, it also requires a little more computational resources and has additional hyper-parameters, which may affect the result. By default, delta node approximation in RxInfer.jl employs the `Unscented` method.
```

**Review by bvdmitri — CHANGES_REQUESTED** [review]
Minor feedback here

File: docs/src/manuals/delta-node.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
+# Delta node manual
+
+RxInfer.jl offers a comprehensive set of stochastic nodes, with a primary emphasis on distributions from the exponential family and its associated compositions, such as GCV or AR nodes. The `DeltaNode` stands out in this package, representing a deterministic transformation of either a single random variable or a group of them. This guide provides insights into the `DeltaNode` and its functionalities.
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:3]:
```suggestion
RxInfer.jl offers a comprehensive set of stochastic nodes, with a primary emphasis on distributions from the exponential family and its associated compositions, such as Gaussian with controlled variance (GCV) or autoregressive (AR) nodes. The `DeltaNode` stands out in this package, representing a deterministic transformation of either a single random variable or a group of them. This guide provides insights into the `DeltaNode` and its functionalities.
```

File: docs/src/manuals/delta-node.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+|---------------|----------------|--------------------------|----------------------
+| Linearization | ✓              | ✗                        | ✓                    
+| Unscented     | ✓              | ✗                        | ✓                    
+| CVI           | ✓              | ✓                        | ✗                    
+
+Based on the above, RxInfer.jl supports the following deterministic transformation scenarios:
+
+1. **Gaussian Nodes**: For delta nodes linked to strictly multivariate or univariate Gaussian distributions, the recommended methods are Linearization or Unscented transforms.
+2. **Exponential Family Nodes**: For the delta node connected to nodes from the exponential family, the CVI (Conjugate Variational Inference) is the method of choice.
+3. **Stacking Delta Nodes**: For scenarios where delta nodes are stacked, either Linearization or Unscented transforms are suitable.
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:19]:
```suggestion
The delta node has several approximation methods for performing probabilistic inference. The desired approximation method depends on the nodes connected to the delta node. We differentiate the following deterministic transformation scenarios:

1. **Gaussian Nodes**: For delta nodes linked to strictly multivariate or univariate Gaussian distributions, the recommended methods are Linearization or Unscented transforms.
2. **Exponential Family Nodes**: For the delta node connected to nodes from the exponential family, the CVI (Conjugate Variational Inference) is the method of choice.
3. **Stacking Delta Nodes**: For scenarios where delta nodes are stacked, either Linearization or Unscented transforms are suitable.

The table below summarizes the features of the delta node in RxInfer.jl, categorized by the approximation method:

| Methods       | Gaussian Nodes | Exponential Family Nodes | Stacking Delta Nodes 
|---------------|----------------|--------------------------|----------------------
| Linearization | ✓              | ✗                        | ✓                    
| Unscented     | ✓              | ✗                        | ✓                    
| CVI           | ✓              | ✓                        | ✗                    
```

File: docs/src/manuals/delta-node.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+## Gaussian Case
+
+In the context of Gaussian distributions, we recommend either the `Linearization` or `Unscented` method for delta node approximation. The `Linearization` method provides a first-order approximation, while the `Unscented` method delivers a more precise second-order approximation. It's worth noting that while the `Unscented` method is more accurate, it also requires a little more computational resources and has additional hyper-parameters, which may affect the result. By default, delta node approximation in RxInfer.jl employs the `Unscented` method.
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:23]: Unscented scales equally good as Linearization in terms of complexity

Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:23]:
```suggestion
In the context of Gaussian distributions, we recommend either the `Linearization` or `Unscented` method for delta node approximation. The `Linearization` method provides a first-order approximation, while the `Unscented` method delivers a more precise second-order approximation. It's worth noting that while the `Unscented` method is more accurate, it also requires a little more computational resources and has additional hyper-parameters, which may affect the result. By default, the delta node approximation employed by RxInfer.jl is the `Unscented` method.
```

File: docs/src/manuals/delta-node.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+or
+```@example delta_node_example
+delta_meta = @meta begin 
+    tanh() -> Unscented()
+end
+```
+
+For a deeper understanding of the [`Unscented`](@ref) method and its parameters, consult the package documentation.
+
+Given the invertibility of `tanh`, indicating its inverse function can optimize inference outcomes:
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:55]:
```suggestion
Given the invertibility of `tanh`, indicating its inverse function can optimize the inference procedure:
```

File: docs/src/manuals/delta-node.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+```@example delta_node_example
+delta_meta = @meta begin 
+    tanh() -> DeltaMeta(method = Linearization(), inverse = atanh)
+end
+```
+
+To execute the inference:
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:63]:
```suggestion
To execute the inference procedure:
```

File: docs/src/manuals/delta-node.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+```julia
+delta_meta = @meta begin 
+    f() -> DeltaMeta(method = Linearization(), inverse=(f_back_x, f_back_g))
+end
+```
+
+## Exponential Family Case
+
+When the delta node is associated with nodes from the exponential family (excluding Gaussians), the `Linearization` and `Unscented` methods are not applicable. In such cases, the CVI (Conjugate Variational Inference) becomes essential. Here's a modified example:
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:109]: You mean when the incoming messages are non-gaussian?

File: docs/src/manuals/delta-node.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+@model function delta_node_example1()
+    z = datavar(Float64)
+    x ~ Gamma(shape=1.0, rate=1.0)
+    y ~ tanh(x)
+    z ~ Bernoulli(y)
+end
+```
+
+The corresponding meta specification is:
```
Comment (bartvanerp) [inline docs/src/manuals/delta-node.md:122]:
```suggestion
The corresponding meta specification can be represented as:
```

File: docs/src/manuals/delta-node.md (hunk 7 above)
Comment (albertpod) [reply docs/src/manuals/delta-node.md:109]: Strictly speaking, that is how it should be phrased. However, I haven't introduced the concept of messages here, so I wanted to refrain from it. Practically "associated" or "connected" to exponential family nodes will require you to use CVI or smth like that.

**Comment by albertpod** [comment]
ping @bvdmitri @bartvanerp.
issues addressed.

**Pull request merged**